core/core_modules/kpm.js

```diff
@@ -220,67 +220,104 @@ var git = require.once('../git.js'),
             }
         });
     },
-    install = function(url, api, event) {
+    installCommon = function (name, moduleLocation, cleanup, api, event) {
+        try {
+            var descriptor = exports.platform.modulesLoader.verifyModule(moduleLocation),
+                moduleList = getModuleList();
+
+            if (!descriptor) {
+                api.sendMessage('"' + name + '" is not a valid module/script.', event.thread_id);
+                cleanup();
+                return;
+            }
+
+            if (exports.platform.loadedModules[descriptor.name] || moduleList[descriptor.name] || moduleList['kpm_' + descriptor.name]) {
+                api.sendMessage('A module with name or directory "' + descriptor.name + '" has already been installed.', event.thread_id);
+                cleanup();
+                return;
+            }
+
+            descriptor.safeName = sanitize(descriptor.name);
+            var instDir = path.resolve('./modules/kpm_' + descriptor.safeName);
+            fs.copy(moduleLocation, instDir, function (err) {
+                if (err) {
+                    console.debug(err);
+                    api.sendMessage('An unknown error occurred while installing "' + descriptor.name + '".', event.thread_id);
+                    cleanup();
+                    return;
+                }
+
+                descriptor.folderPath = instDir;
+                var m = exports.platform.modulesLoader.loadModule(descriptor);
+                if (m !== null) {
+                    moduleCache[descriptor.name] = descriptor;
+                    exports.platform.loadedModules.push(m);
+                    api.sendMessage('"' + descriptor.name + '" (' + descriptor.version + ') is now installed.', event.thread_id);
+                }
+                else {
+                    api.sendMessage('"' + descriptor.name + '" (' + descriptor.version + ') could not be installed, it appears to be invalid (syntax error?).', event.thread_id);
+                    fs.emptyDir(descriptor.folderPath, function () {
+                        // just delete if we can, not a lot we can do about errors here.
+                    });
+                }
+                cleanup();
+            });
+        }
+        catch (e) {
+            console.critical(e);
+            api.sendMessage('Could not install "' + name + '".', event.thread_id);
+            cleanup();
+        }
+    },
+
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["gitInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976242/issues/source?bid=3282987&fileBranchId=3282998#l273)